ci/scripts/gen-integration-test-yaml.py

```diff
@@ -77,6 +77,9 @@ def format_pipeline_yaml_cmd(pipeline_steps):
     # Agent terminated because the AWS EC2 spot instance killed by AWS.
     - signal_reason: agent_stop
       limit: 3
+    - exit_status: -1
+      signal_reason: none
+      limit: 3
```

Comment:
I found discussion here: https://github.com/risingwavelabs/risingwave/pull/11345#discussion_r1280315011

Reply:
IIUC the idea is we don't want to retry timeout'd tests. (BTW, would -1 also occur for other bash errors?)

Reply:
The timeouts I observed recently were all with exit code 143, which is `SIGTERM`.

Even for the case mentioned in https://github.com/risingwavelabs/risingwave/pull/11305#issuecomment-1656004765, there's a `signal_reason` of `SIGKILL`, while we only match `none` here.

Reply:
buildkite.com/risingwavelabs/main-cron/builds/4973#019653ee-6c34-45fc-8f1b-41218534b599


This sais Timeout with status -1. Did I miss sth?
<img width="1319" alt="image" src="https://github.com/user-attachments/assets/3b39ec53-a53a-44e1-8455-c5b85cd18036" />

Reply:
It's `1` instead of `-1`, and I believe it's because of this error:

<img width="1144" alt="image" src="https://github.com/user-attachments/assets/bdb3ba31-563c-41f7-82b1-c164dfa6a101" />

Reply:
🤣

Reply:
Oh! I think `signal_reason: none` plus `-1` can mean "agent lost"?

Reply:
none: No signal was sent to the process.

Reply:
Oh, according to the doc https://buildkite.com/docs/agent/v3#exit-codes
The exit code of the script will be transformed by the buildkite agent

Reply:
> I believe it's because of this error (no `junit` xml found)

Seems it's because sqllogictest did not exit gracefully in 30 seconds, so there's no chance to dump the junit file.